base/applications/mspaint/winproc.cpp

```diff
@@ -275,53 +275,59 @@ LRESULT CMainWindow::OnClose(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHan
     return 0;
 }
 
+void CMainWindow::ProcessFileMenu(HMENU hPopupMenu)
+{
+    LPCTSTR dotext = PathFindExtensionW(filepathname);
+    BOOL isBMP = FALSE;
+    if (_tcsicmp(dotext, _T(".bmp")) == 0 ||
+        _tcsicmp(dotext, _T(".dib")) == 0 ||
+        _tcsicmp(dotext, _T(".rle")) == 0)
+    {
+        isBMP = TRUE;
+    }
+
+    EnableMenuItem(hPopupMenu, IDM_FILEASWALLPAPERPLANE,     ENABLED_IF(isAFile && isBMP));
+    EnableMenuItem(hPopupMenu, IDM_FILEASWALLPAPERCENTERED,  ENABLED_IF(isAFile && isBMP));
+    EnableMenuItem(hPopupMenu, IDM_FILEASWALLPAPERSTRETCHED, ENABLED_IF(isAFile && isBMP));
+
+    for (INT iItem = 0; iItem < MAX_RECENT_FILES; ++iItem)
+        RemoveMenu(hPopupMenu, IDM_FILE1 + iItem, MF_BYCOMMAND);
+
+    if (registrySettings.strFiles[0].IsEmpty())
+        return;
+
+    RemoveMenu(hPopupMenu, IDM_FILEMOSTRECENTLYUSEDFILE, MF_BYCOMMAND);
+
+    INT cItems = GetMenuItemCount(hPopupMenu);
+
+    for (INT iItem = 0; iItem < MAX_RECENT_FILES; ++iItem)
+    {
+        CString& strFile = registrySettings.strFiles[iItem];
+        if (strFile.IsEmpty())
+            break;
+
+        CPath pathFile(strFile);
+        pathFile.CompactPathEx(40);
```

Comment:
A lot of magic numbers in here. Is 40 now, was 30 before. Why?
I do assume that the wsprintf at line 313 should not overwrite the szText buffer. Is that 100% sure currently?
Do we already have strsafe included here? If yes, then maybe we should use that insead.
If not included already, then maybe someone else who is better at counting digits than my brain can confirm that the wsprintf is guaranteed here to not overflow the buffer.

Reply:
> Is 40 now, was 30 before. Why?

Reverted to 30.

> should not overwrite the szText buffer. Is that 100% sure currently?

Added an assertion.

> Do we already have strsafe included here?

No.